cmake/compile_definitions/common.cmake

```diff
@@ -56,8 +56,37 @@ set(SUNSHINE_TARGET_FILES
         "${CMAKE_SOURCE_DIR}/third-party/moonlight-common-c/src/RtspParser.c"
         "${CMAKE_SOURCE_DIR}/third-party/moonlight-common-c/src/Video.h"
         "${CMAKE_SOURCE_DIR}/third-party/tray/src/tray.h"
-        "${CMAKE_SOURCE_DIR}/src/upnp.cpp"
-        "${CMAKE_SOURCE_DIR}/src/upnp.h"
+        "${CMAKE_SOURCE_DIR}/src/input/common.cpp"
```

Comment:
Maybe this is a good opportunity to use a glob?

Also, it's not your doing, but I believe `SUNSHINE_TARGET_FILES` is only supposed to be `c/cpp`. And `h` files are supposed to be added here: https://github.com/LizardByte/Sunshine/blob/f0a00ae35683dfd5742952fb4cf99edf268b57fd/cmake/targets/common.cmake#L25 (under a different variable name)

Perhaps we could use `SUNSHINE_SOURCES` for c/cpp and `SUNSHINE_HEADERS` for the headers?

I was also wondering if our headers should be renamed to have the `hpp` extension. This seems to be common for cpp projects.

Reply:
Yes, how to I do glob here?

I'm not sure, I don't really know enough about the linking and stuff atm, can we have opinion from more people on it? Maybe we should open a PR for that. Regarding the `hpp`, I'm not sure, from my understanding it supports some templating stuff that `h` don't, but other than that it's just preference, I'd just need to confirm it won't break Objective-C interop we use on macOS.

Reply:
> Yes, how to I do glob here?

Here's an example: https://github.com/LizardByte/Sunshine/blob/f0a00ae35683dfd5742952fb4cf99edf268b57fd/tests/CMakeLists.txt#L96

And yea it could be done in a separate PR.

Regarding headers, I'm also no expert on it. Just thought about it when reviewing this PR since it touches a lot of files. https://stackoverflow.com/questions/152555/h-or-hpp-for-your-class-definitions

Can also be addressed later though.

Reply:
Just for now, included a glob `src/** !platform` and we improve later